### ISSUE TITLE
Enhancement: make debugger respect user-defined extension settings

### ISSUE THREAD

**Issue by Antonius-S** [Microsoft/vscode-node-debug2#228]
Currently extension doesn't respect user-defined non-standard extension settings neither from `files.associations` nor `\VSCode\resources\app\extensions\javascript\package.json`. Would be nice if it did.
Goal: to have an ability to launch JS files with non-standard extension, f.ex. `*.njs` or `*.node` (to distinguish Node sources from browser JS and shell JS).

**Comment by roblourens** [comment]
Sorry, I don't quite understand what you are asking for.

**Comment by Antonius-S** [comment]
@roblourens now debugger refuses to launch a file with non-standard extension with error "setting outFiles might help". The list of supported file extensions is hard-coded. I suggest extension to pick user-defined extensions from settings or at least from resources file.

**Comment by roblourens** [comment]
Got it, we should be able to consider `files.associations`. Do you mean `.mjs`? Or are you using your own extension just to indicate that the file runs in node?

**Comment by Antonius-S** [comment]
Yep, you got the idea. I use my own extension for Node projects to associate them with Node, differentiate from browser JS and WSH scripts etc.

**Comment by weinand** [comment]
@roblourens instead of hardcoding the JS extensions, we could pass them as a private property on the `launch` or `attach` requests from the extension to the DA. The extension has access to everything that is needs, e.g. the files.associations.
